TreeMaker/python/makeTreeFromMiniAOD_cff.py

```diff
@@ -979,22 +979,41 @@ def makeTreeFromMiniAOD(self,process):
     ## ----------------------------------------------------------------------------------------------
     ## Prefiring weights
     ## ----------------------------------------------------------------------------------------------
-    from TreeMaker.Utils.L1ECALNonPrefiringProbProducer_cfi import L1ECALNonPrefiringProbProducer
-    prefiringDataEra = cms.PSet(value = cms.string(""))
-    TMeras.TM2016.toModify(prefiringDataEra, value = "2016BtoH")
-    TMeras.TM2017.toModify(prefiringDataEra, value = "2017BtoF")
-    if len(prefiringDataEra.value.value())>0:
-        process.L1ECALNonPrefiringProbProducer = L1ECALNonPrefiringProbProducer.clone(
+    from TreeMaker.Utils.L1NonPrefiringProbProducer_cfi import L1NonPrefiringProbProducer
+    prefiringDataEraECAL = cms.PSet(value = cms.string(""))
+    prefiringDataEraMuon = cms.PSet(value = cms.string(""))
+
+    TMeras.TMUL2016.toModify(prefiringDataEraECAL,    value = "UL2016preVFP")
+    TMeras.TMUL2016APV.toModify(prefiringDataEraECAL, value = "UL2016postVFP")
+    TMeras.TMUL2017.toModify(prefiringDataEraECAL,    value = "UL2017BtoF")
+    TMeras.TMUL2018.toModify(prefiringDataEraECAL,    value = "None")
+
+    TMeras.TMUL2016.toModify(prefiringDataEraMuon,    value = "2016preVFP")
+    TMeras.TMUL2016APV.toModify(prefiringDataEraMuon, value = "2016postVFP")
+    TMeras.TMUL2017.toModify(prefiringDataEraMuon,    value = "20172018")
+    TMeras.TMUL2018.toModify(prefiringDataEraMuon,    value = "20172018")
+
+    if len(prefiringDataEraECAL.value.value())>0 and len(prefiringDataEraMuon.value.value())>0:
```

Comment:
rather than extra PSets and if statements, just create `process.L1NonPrefiringProbProducer` with default values and then apply `TMeras.TMUL2018.toModify(process.L1NonPrefiringProbProducer, ...)` directly.

Reply:
Yeah, will simplify.